client/src/api/request/index.js

```diff
@@ -0,0 +1,28 @@
+import jsonFetch from "simple-json-fetch";
+
+/**
+ * Request
+ * @param  {string} requestUrl url end_point
+ * @param  {object} requestParameters
+ * @param  {func} callback
+ */
+const request = (requestUrl = null, requestParameters, callback) => {
+  if (null == requestUrl) {
+    throw new Error(`url ${requestUrl} malformed `);
+  }
+  if ("function" != typeof callback) {
+    throw new Error(`callback - ${callback} - is not a function`);
+  }
+  return dispatch => {
+    return jsonFetch(requestUrl, requestParameters).then(
+      response => {
+        dispatch(callback(response, requestParameters));
+      },
+      response => {
+        dispatch(callback(response, requestParameters, true));
```

Comment:
Ah, this is the error callback. I guess we're using this `true` to mean it's an error or something?

Reply:
yes exactly, `jsonFetch` returns a `Promise((resolve, reject)`. So I pass a boolean in my callback. false: no error,  true: errors